macros.cmake

```diff
@@ -231,7 +231,7 @@ macro(register_derived_artifact FROM_ARTIFACTS NAME SUFFIX TYPE)
             add_custom_target(
               compile-${NAME}_${ARTIFACT_CATEGORY}__${ARTIFACT}__${CONFIGURATION}
               ALL
-              DEPENDS ${OUTPUT} ${DEPEND_ON})
+              DEPENDS ${OUTPUT} ${DEPEND_ON} generated_abi_artifact_sources)
```

Comment:
You could create a variable to define the variable and define it to an empty string if not revng-qa.

Reply:
Yeah, I see your point.

But that's not the main issue I'm concerned about. Having this dependency outside of `revng-qa` is not really a problem since creating an "empty" target is pretty much free.

What I hate is the idea that this forces totally unrelated targets (all of them, to be exact) to depend on a single script they don't even need to depend on - de facto creating a bottleneck - no target can start building until the script is over.

Reply:
`revng-qa` currently takes no time to build. Let's not optimize it if it's not broken.
Also, I'm trying to re-engineer all of this from scratch, so no need to make it good.

Reply:
I'm OK with either options. Important part is that we don't use `execute_process`/generate stuff in the build directory.

Squash squash squash